### PR TITLE
fix invalid scheduled tx on epoch start

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
- fix invalid scheduled tx on epoch start
- fixed transaction restore in cache
- added unit tests

### CODE REVIEW

File: process/block/preprocess/scheduledTxsExecution.go [hunk 1]
```diff
@@ -268,6 +268,15 @@ func (ste *scheduledTxsExecution) removeInvalidTxsFromScheduledMiniBlocks(interm
 		}
 	}
 
+	resultedSlice := make(block.MiniBlockSlice, 0)
```
Comment (SebastianMarian) [inline process/block/preprocess/scheduledTxsExecution.go:271]: resultedScheduledMbs

File: process/block/preprocess/transactions.go [hunk 2]
```diff
@@ -253,7 +253,7 @@ func (txs *transactions) RestoreBlockDataIntoPools(
 			continue
 		}
 
-		strCache := process.ShardCacherIdentifier(miniBlock.SenderShardID, miniBlock.ReceiverShardID)
+		miniblockStrCache := process.ShardCacherIdentifier(miniBlock.SenderShardID, miniBlock.ReceiverShardID)
```
Comment (SebastianMarian) [inline process/block/preprocess/transactions.go:256]: mini**B**lockStrCache

Comment (sasurobert) [reply process/block/preprocess/transactions.go:256]: +1

File: process/block/preprocess/scheduledTxsExecution.go (hunk 1 above)
Comment (iulianpascalau) [reply process/block/preprocess/scheduledTxsExecution.go:271]: renamed

File: process/block/preprocess/transactions.go (hunk 2 above)
Comment (iulianpascalau) [reply process/block/preprocess/transactions.go:256]: done

**Review by miiu96 — DISMISSED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**